src/path.cr

```diff
@@ -0,0 +1,847 @@
+  #
+  # ```
+  # Path.windows("C:\Program Files").drive    # => Path.windows("C:")
+  # Path.windows("\\host\share\folder").drive # => Path.windows("\\host\share")
+  # ```
+  #
+  # NOTE: Drives are only available for Windows paths. It can either be a drive letter (`C:`) or a UNC share (`\\host\share`).
+  def drive : Path?
+    if drive = @type.drive(@name)
+      new_instance drive
+    end
+  end
+
+  # Returns the root path component of this path or `nil` if it is not rooted.
+  #
+  # See `#anchor` for the combination of `#drive` and root.
+  #
+  # ```
+  # Path.posix("/etc/").root                 # => Path.posix("/")
+  # Path.windows("C:Program Files").root     # => nil
+  # Path.windows("C:\Program Files").root    # => Path.windows("\")
+  # Path.windows("\\host\share\folder").root # => Path.windows("\")
+  # ```
+  def root : Path?
+    if root = @type.root(@name)
+      new_instance root
+    end
+  end
+
+  # Returns the concatenation of `#drive` and `#root`.
+  #
+  # ```
+  # Path.posix("/etc/").anchor                 # => Path.posix("/")
+  # Path.windows("C:Program Files").anchor     # => Path.windows("C:")
+  # Path.windows("C:\Program Files").anchor    # => Path.windows("C:\")
+  # Path.windows("\\host\share\folder").anchor # => Path.windows("\\host\share\")
+  # ```
+  def anchor : Path?
+    root = root()
+    return root if posix?
+
+    drive = drive()
+    if drive
+      if root
+        drive.join(root)
+      else
+        drive
+      end
+    else
+      root
+    end
+  end
+
+  # Returns `true` if this path is absolute.
+  #
+  # A POSIX path is absolute if it begins with a forward slash (`/`).
+  # A Windows path is absolute if it begins with a drive letter and root (`C:\`)
+  # or with a UNC share (`\\server\share\`).
+  def absolute? : Bool
+    @type.absolute?(@name)
+  end
+
+  def ends_with_separator?
+    @type.separators.includes?(@name[-1]?)
+  end
+
+  # Returns the string representation of this path.
+  def to_s : String
+    @name
+  end
+
+  # Appends the string representation of this path to *io*.
+  def to_s(io : IO)
+    io << to_s
+  end
+
+  # Inspects this path to *io*.
+  def inspect(io : IO)
+    if native?
+      io << "Path["
+      @name.inspect(io)
+      io << ']'
+    else
+      io << "Path."
+      io << (windows? ? "windows" : "posix")
+      io << '('
+      @name.inspect(io)
+      io << ')'
+    end
+  end
+
+  # Returns a new `URI` with `file` scheme from this path.
+  #
+  # A URI can only be created with an absolute path. Raises `Path::Error` if
+  # this path is not absolute.
+  def to_uri : URI
+    raise Error.new("Cannot create a URI from relative path") unless absolute?
+    URI.new(scheme: "file", path: @name)
+  end
+
+  # Returns the current working directory of the process.
+  def self.current : Path
+    new Crystal::System::Dir.current
+  end
+
+  # Changes the current working directory of the process to the given path.
+  def self.current=(path : Path | String)
```

Comment:
1. I *really* dislike a setter having side effects.
2. `Path` is purely theoretical.
3. Duplicates `Dir.cd(path)`.

Having `Path.current` returning a `Path` of the current directory is enough. Let's keep changing the current working directory in `Dir`, please.

Reply:
Sounds legit. But if we remove it, I think getter `Path.current` should also be removed. It's essentially an alias for `Dir.current` and with `Dir.cd` it makes sense to keep everything related to the working directory in `Dir`.